src/registry.ts

```diff
@@ -20,7 +20,7 @@ class Registry implements IRegistry {
 
   emit(name: string, ...args: any) {
     this.heartbeat = Math.max(new Date().getTime(), this.heartbeat);
-    this.emit(name, ...args);
+    this.events.emit(name, ...args);
```

Comment:
Given you're fixing this function was calling itself, maybe it's a good time to introduce a test for `emit`